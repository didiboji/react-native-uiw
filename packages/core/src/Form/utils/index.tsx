import cloneDeepWith from 'lodash/cloneDeepWith';
import { isObject, isArray } from './is';

export function cloneDeep(value: any) {
  // 只有对象才执行拷贝，否则直接返回。 如果是 File，MouseEvent对象，都可以直接返回
  return cloneDeepWith(value, (val) => {
    if (!isObject(val) && !isArray(val)) {
      return val;
    }
  });
}
