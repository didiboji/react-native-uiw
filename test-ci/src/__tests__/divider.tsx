import renderer from 'react-test-renderer';

import 'react-native';
import React from 'react';
import Divider from '../lib/Divider';
import { render } from '@testing-library/react-native';
import { keys, toObject } from '../utils';

const TYPES: keys = {
  horizontal: 'row',
  vertical: 'column',
};

describe('Divider', () => {
  it('label', () => {
    const { getByTestId } = render(<Divider label="这是一条分割线" />);
    const component = getByTestId('RNE__Divider__label');
    const lable = component.props.children[1];
    expect(lable).toBe('这是一条分割线');
  });

  describe.each`
    type
    ${'horizontal'}
    ${'vertical'}
  `('$type', ({ type }) => {
    it(`should display type ${type} divider`, () => {
      const { getByTestId } = render(<Divider type={type} />);
      const component = getByTestId('RNE__Divider__wrap');
      const styles = toObject(component.props.style);
      expect(styles.flexDirection).toBe(TYPES[type]);
    });
  });

  it('gutter', () => {
    const { getByTestId } = render(<Divider gutter={12} />);
    const component = getByTestId('RNE__Divider__start');
    const styles = toObject(component.props.style);
    expect(styles.marginHorizontal).toBe(12);
  });

  describe.each`
    orientation
    ${'left'}
    ${'right'}
    ${'center'}
  `('$orientation', ({ orientation }) => {
    it(`should display orientation ${orientation} divider`, () => {
      const { toJSON } = render(<Divider label="这是一条分割线" orientation={orientation} />);
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
