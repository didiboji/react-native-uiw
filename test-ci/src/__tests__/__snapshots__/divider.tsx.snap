// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider center should display orientation center divider 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
      },
      {
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  testID="RNE__Divider__wrap"
>
  <View
    style={
      [
        {
          "backgroundColor": "rgb(229, 229, 229)",
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 100,
        },
        {
          "marginHorizontal": 8,
        },
        {
          "height": 1,
          "minHeight": 1,
          "minWidth": 16,
        },
        undefined,
      ]
    }
    testID="RNE__Divider__start"
  />
  <Text
    style={
      [
        {
          "color": "rgba(0, 0, 0, 0.54)",
          "fontSize": 14,
        },
        undefined,
      ]
    }
    testID="RNE__Divider__label"
  >
     
    这是一条分割线
     
  </Text>
  <View
    style={
      [
        {
          "backgroundColor": "rgb(229, 229, 229)",
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 100,
        },
        {
          "marginHorizontal": 8,
        },
        {
          "height": 1,
          "minHeight": 1,
          "minWidth": 16,
        },
        undefined,
      ]
    }
    testID="RNE__Divider__end"
  />
</View>
`;

exports[`Divider left should display orientation left divider 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
      },
      {
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  testID="RNE__Divider__wrap"
>
  <View
    style={
      [
        {
          "backgroundColor": "rgb(229, 229, 229)",
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 100,
        },
        {
          "marginHorizontal": 8,
        },
        {
          "height": 1,
          "minHeight": 1,
          "minWidth": 16,
        },
        undefined,
      ]
    }
    testID="RNE__Divider__start"
  />
  <Text
    style={
      [
        {
          "color": "rgba(0, 0, 0, 0.54)",
          "fontSize": 14,
        },
        undefined,
      ]
    }
    testID="RNE__Divider__label"
  >
     
    这是一条分割线
     
  </Text>
  <View
    style={
      [
        {
          "backgroundColor": "rgb(229, 229, 229)",
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 100,
        },
        {
          "marginHorizontal": 8,
        },
        {
          "height": 1,
          "minHeight": 1,
          "minWidth": 16,
        },
        {
          "flexGrow": 100,
        },
        undefined,
      ]
    }
    testID="RNE__Divider__end"
  />
</View>
`;

exports[`Divider right should display orientation right divider 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
      },
      {
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  testID="RNE__Divider__wrap"
>
  <View
    style={
      [
        {
          "backgroundColor": "rgb(229, 229, 229)",
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 100,
        },
        {
          "marginHorizontal": 8,
        },
        {
          "height": 1,
          "minHeight": 1,
          "minWidth": 16,
        },
        {
          "flexGrow": 100,
        },
        undefined,
      ]
    }
    testID="RNE__Divider__start"
  />
  <Text
    style={
      [
        {
          "color": "rgba(0, 0, 0, 0.54)",
          "fontSize": 14,
        },
        undefined,
      ]
    }
    testID="RNE__Divider__label"
  >
     
    这是一条分割线
     
  </Text>
  <View
    style={
      [
        {
          "backgroundColor": "rgb(229, 229, 229)",
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 100,
        },
        {
          "marginHorizontal": 8,
        },
        {
          "height": 1,
          "minHeight": 1,
          "minWidth": 16,
        },
        undefined,
      ]
    }
    testID="RNE__Divider__end"
  />
</View>
`;
